tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -460,5 +497,169 @@ private static bool ParseBooleanConfigurationValue(string value)
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if (displayName is null)
+                {
+                    Utils.WriteError(TraceProgramNotFound);
+                }
+                else if (displayName is datadog64BitProgram)
+                {
+                    Utils.WriteSuccess(TracerProgramFound(datadog64BitProgram));
+                }
+                else if (displayName is datadog32BitProgram)
+                {
+                    if (RuntimeInformation.OSArchitecture is Architecture.X64)
+                    {
+                        Utils.WriteError(WrongTracerArchitecture(datadog32BitProgram));
+                        return tracerVersion;
+                    }
+
+                    Utils.WriteSuccess(TracerProgramFound(datadog32BitProgram));
+                }
+            }
+
+            return tracerVersion;
+        }
+
+        private static void CheckAppPools(ApplicationPool? pool)
+        {
+            var tracingRelevantVariables = new Dictionary<string, string>
+            {
+                { "COR_ENABLE_PROFILING", "1" },
+                { "CORECLR_ENABLE_PROFILING", "1" },
+                { "COR_PROFILER", Utils.Profilerid },
+                { "CORECLR_PROFILER", Utils.Profilerid }
+            };
+
+            var variableFound = false;
+            var foundVariables = new Dictionary<string, object>();
+
+            if (pool != null)
+            {
+                var environmentVariablesCollection = pool.GetCollection("environmentVariables");
+                foreach (var variable in environmentVariablesCollection)
+                {
+                    var name = (string)variable.Attributes["name"].Value;
+                    var value = (string)variable.Attributes["value"].Value;
+
+                    if (tracingRelevantVariables.ContainsKey(name) && value != tracingRelevantVariables[name])
+                    {
+                        variableFound = true;
+                        foundVariables[name] = value;
+                    }
+                }
+
+                if (variableFound)
+                {
+                    Utils.WriteWarning(AppPoolCheckFindings(pool.Name));
+
+                    foreach (var variable in foundVariables)
+                    {
+                        Utils.WriteError(WrongEnvironmentVariableFormat(variable.Key, tracingRelevantVariables[variable.Key], variable.Value.ToString()));
+                    }
+
+                    foundVariables.Clear();
+                    variableFound = false;
+                }
+            }
+
+            var serverManager = new ServerManager();
+            var defaultEnvironmentVariables = serverManager.ApplicationPools.GetChildElement("applicationPoolDefaults").GetCollection("environmentVariables");
```

Comment:
Might be nice to add some unit tests around this, by creating an interface and implementation, similar to `IRegistryService`. Not definitely required, as the logic is pretty simple I think, just an option